### PR TITLE
FilesystemStorageActor a.k.a. thread safety for file CAS

### PR DESCRIPTION
**Pull request by thedavidmeister**
i believe this _should_ make the file backed CAS safe to use across threads

there is an actor under the hood
the actor reference re-uses existing actor reference paths in riker if the same dir is called
the actor reference is the new internal state of `FilesystemStorage` rather than the path

### CODE REVIEW

File: core/src/cas/file/actor.rs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        Ok(write(self.address_to_path(address), content)?)
+    }
+
+    /// filesystem CAS contains. NOT thread safe.
+    fn unsafe_contains(&self, address: &Address) -> Result<bool, HolochainError> {
+        Ok(Path::new(&self.address_to_path(address)).is_file())
+    }
+
+    /// filesystem CAS fetch. NOT thread safe.
+    fn unsafe_fetch(&self, address: &Address) -> Result<Option<Content>, HolochainError> {
```
Comment (AshantiMutinta) [inline core/src/cas/file/actor.rs]: I was afraid we were using unsafe rust and my eyes popped out haha. I think the naming gives the impression that we are using a subsection of rust that behaves in an unsafe context https://doc.rust-lang.org/book/second-edition/ch19-01-unsafe-rust.html

Comment (thedavidmeister) [reply core/src/cas/file/actor.rs]: @AshantiMutinta that's fair. would you accept `unthreadable_fetch`?

Comment (AshantiMutinta) [reply core/src/cas/file/actor.rs]: unthreadable is good but then again we have to ask what properties does this have which make them so unthreadable. I guess it is the lack of the ARC for the context? In that case I think that is a good name. Just highlighting the question of, if I had to put this in a thread::spawn, what would stop me? Is it unthreadable or unthreaded?

Comment (ddd-mtl) [reply core/src/cas/file/actor.rs]: Random suggestion `single_threaded_fetch()` or  `st_fetch()`

**Pull request merged**